src/demo/java/org/junitpioneer/jupiter/EnvironmentVariablesExtensionDemo.java

```diff
@@ -26,7 +27,8 @@
 import org.junit.jupiter.params.provider.ValueSource;
 import org.junitpioneer.jupiter.params.IntRangeSource;
 
-@EnabledForJreRange(max = JRE.JAVA_16, disabledReason = "See: https://github.com/junit-pioneer/junit-pioneer/issues/509")
+@Disabled
+@EnabledForJreRange(max = JRE.JAVA_17, disabledReason = "See: https://github.com/junit-pioneer/junit-pioneer/issues/509")
```

Comment:
Can't this be fixed by configuring Gradle's `test` task with `--add-opens`?

Reply:
Created #853 for it, to get this big change (including all the documentation things) through the door.

_Yes, maybe Java 17 and JUnit 6 should have been split. On the other side they are quite close together. Don't know._